### ISSUE TITLE
Explicitly add conversion to Matrix

### ISSUE THREAD

**Issue by kescobo** [JuliaData/DataFrames.jl#1474]
Constructors no longer fall back to `convert()` on 0.7, so `Matrix(df)` now throws an error. 

julia0.6:

```
julia> using DataFrames

julia> df = DataFrame(a=[1,2], b=[3,4])
2×2 DataFrames.DataFrame
│ Row │ a │ b │
├─────┼───┼───┤
│ 1   │ 1 │ 3 │
│ 2   │ 2 │ 4 │

julia> Matrix(df)
2×2 Array{Int64,2}:
 1  3
 2  4
```

julia0.7

```
julia> using DataFrames

julia> df = DataFrame(a=[1,2], b=[3,4])
2×2 DataFrame
│ Row │ a │ b │
├─────┼───┼───┤
│ 1   │ 1 │ 3 │
│ 2   │ 2 │ 4 │

julia> Matrix(df)
ERROR: MethodError: no method matching Array{T,2} where T(::DataFrame)
#...
```

This can be replaced with `convert(Matrix, df)`, but it might be worth adding this functionality back (as discussed with @nalimilan on Slack).

From the REPL, it seems like it might be as easy as 

```
julia> import Base.Matrix

julia> @inline Matrix(df::DataFrame) = convert(Matrix, df)
Array{T,2} where T

julia> Matrix(df)
2×2 Array{Int64,2}:
 1  3
 2  4
```

If so, I'm happy to submit a PR if that would be helpful (I'd just want to know what the best place in the code base for the method would be). But if it's more complicated it's perhaps more efficiently handled by someone that knows more.

**Comment by bkamins** [comment]
I think it is OK to add `Matrix` and `Matrix{T}` as they are both possibly convenient. I would just add them next to `convert` methods.

**Comment by nalimilan** [comment]
A PR would be welcome. No need for `@inline` though.

**Comment by kescobo** [comment]
>I think it is OK to add Matrix and Matrix{T} as they are both possibly convenient. I would just add them next to convert methods.

@bkamins is the latter necessary? Just now in 0.7 I did:

```
julia> using DataFrames
[ Info: Recompiling stale cache file /Users/kev/.julia/compiled/v0.7/DataFrames/AR9oZ.ji for DataFrames [a93c6f00-e57d-5684-b7b6-d8193f3e46c0]
┌ Warning: `using A.B` will only be allowed for modules, not single bindings. Use `using A: B` instead
│   caller = ip:0x0
└ @ Core :-1

julia> Base.Matrix(df::AbstractDataFrame) = Base.convert(Matrix, df)

julia> df = DataFrame(a=[1], b=[2])
1×2 DataFrame
│ Row │ a │ b │
├─────┼───┼───┤
│ 1   │ 1 │ 2 │

julia> Matrix(df)
1×2 Array{Int64,2}:
 1  2

julia> Matrix{Float64}(df)
1×2 Array{Float64,2}:
 1.0  2.0
```

@nalimilan Thanks - If I'm being honest I don't 100% understand what `@inline` does 😕... I've just seen it in a bunch of similar places

**Comment by bkamins** [comment]
I can see that in the PR you have added `Matrix{T}` constructor and I think it is needed. Does it really work for you without it?

**Comment by kescobo** [comment]
@bkamins It does seem to be necessary - if I remove it those constructors fail 🤷‍♀️

**Comment by kescobo** [comment]
No sweat! I think I must have already added the code to the repo, and then opened the terminal when I tried before (so the constructor was already defined). I just checked out master and tried the exact same thing from the REPL and got an error.

**Comment by nalimilan** [comment]
Fixed by #1475.
